File: app/src/main/java/ru/evotor/framework/core/action/event/inventory/ProductCardOpenedEvent.java
package ru.evotor.framework.core.action.event.inventory;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import ru.evotor.framework.core.action.event.cash_operations.CashOutEvent;

public class ProductCardOpenedEvent extends ProductEvent {
    public static final String BROADCAST_ACTION_PRODUCT_CARD_OPENED = "evotor.intent.action.inventory.CARD_OPEN";

    public ProductCardOpenedEvent(@Nullable String productUuid) {
        super(productUuid);
    }

    private ProductCardOpenedEvent(@NonNull Bundle extras) {
        super(extras);
    }

    @Nullable
    public static ProductCardOpenedEvent create(@Nullable Bundle bundle) {
        if (bundle == null) {
            return null;
        }
        return new ProductCardOpenedEvent(bundle);
    }
}

File: src/main/java/ru/evotor/framework/inventory/event/ProductCardOpenedEvent.kt
package ru.evotor.framework.inventory.event

import android.os.Bundle

/**
 * Событие открытия карточки товара.
 *
 * Происходит при открытии карточки товара в интерфейсе приложения "Товары".
 *
 * Обрабатывать это событие можно с помощью широковещательного приёмника товароучётных событий:
 * [ru.evotor.framework.inventory.event.handler.receiver.InventoryBroadcastReceiver]
 *
 * @param productUuid uuid товара
 */
class ProductCardOpenedEvent(productUuid: String) : ProductEvent(productUuid) {
    companion object {
        fun from(bundle: Bundle?): ProductCardOpenedEvent? = bundle?.let {
            ProductCardOpenedEvent(ProductEvent.getProductUuid(it) ?: return null)
        }
    }
}